py-polars/polars/internals/dataframe/frame.py

```diff
@@ -6504,9 +6530,25 @@ def iterrows(
         >>> [row.b for row in df.iterrows(named=True)]
         [2, 4, 6]
 
-        """
-        if named:
-            Row = namedtuple("Row", self.columns)  # type: ignore[misc]
+        See Also
+        --------
+        rows : materialises all frame data as a list of rows.
+
+        """
+        # note: buffering rows results in a 2-4x speedup over individual calls
+        # to ".row(i)", so it should only be disabled in extremely specific cases.
+        if buffer_size:
+            for offset in range(0, self.height, buffer_size):
+                rows_chunk = self.slice(offset, buffer_size).rows(named=False)
+                if named:
+                    Row = namedtuple("Row", self.columns)  # type: ignore[misc]
+                    for row in rows_chunk:
+                        yield Row(*row)
+                else:
+                    yield from rows_chunk
+
+        elif named:
+            Row = namedtuple("Row", self.columns)  # type: ignore[misc,no-redef]
```

Comment:
I think it's a bit cleaner to define this namedtuple once at  the top of this function, and then use that. See 6406 for what I mean.

Reply:
That was quick 😅 personally I put the declaration behind an `if named:` statement, but creating the tuple should not be too expensive.

Reply:
Quite right! I had moved the existing code down, and then despite the very obvious mypy hint that I was redeclaring `Row` I forgot to refactor it up, heh. Tweaked accordingly :)